buf/registry/plugin/v1beta1/plugin_service.proto

```diff
@@ -0,0 +1,217 @@
+  rpc GetPlugins(GetPluginsRequest) returns (GetPluginsResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // List Plugins, usually for a specific Organization.
+  rpc ListPlugins(ListPluginsRequest) returns (ListPluginsResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // Create new Plugins.
+  //
+  // This operation is atomic. Either all Plugins are created or an error is returned.
+  rpc CreatePlugins(CreatePluginsRequest) returns (CreatePluginsResponse) {
+    option idempotency_level = IDEMPOTENT;
+  }
+  // Update existing Plugins.
+  //
+  // This operation is atomic. Either all Plugins are updated or an error is returned.
+  rpc UpdatePlugins(UpdatePluginsRequest) returns (UpdatePluginsResponse) {
+    option idempotency_level = IDEMPOTENT;
+  }
+  // Delete existing Plugins.
+  //
+  // This operation is atomic. Either all Plugins are deleted or an error is returned.
+  rpc DeletePlugins(DeletePluginsRequest) returns (DeletePluginsResponse) {
+    option idempotency_level = IDEMPOTENT;
+  }
+}
+
+message GetPluginsRequest {
+  // The Plugins to request.
+  repeated PluginRef plugin_refs = 1 [
+    (buf.validate.field).repeated.min_items = 1,
+    (buf.validate.field).repeated.max_items = 250
+  ];
+}
+
+message GetPluginsResponse {
+  // The retrieved Plugins in the same order as requested.
+  repeated Plugin plugins = 1 [(buf.validate.field).repeated.min_items = 1];
+}
+
+message ListPluginsRequest {
+  // The list order.
+  enum Order {
+    ORDER_UNSPECIFIED = 0;
+    // Order by create_time newest to oldest.
+    ORDER_CREATE_TIME_DESC = 1;
+    // Order by create_time oldest to newest.
+    ORDER_CREATE_TIME_ASC = 2;
+  }
+  // Filter for the list request, if any. When multiple filters are specified, they are combined.
+  message Filter {
+    // The specific Organizations to list Plugins for.
+    //
+    // If empty, all Plugins for all Organizations are listed, but this functionality is limited to
+    // Users with the necessary permissions.
+    repeated buf.registry.owner.v1.OwnerRef owner_refs = 1;
+    // The collections to list Plugins for.
+    //
+    // If empty, all Plugins are listed regardless of collection.
+    repeated string collection_names = 2;
+    // The languages to list Plugins for.
+    //
+    // If empty, all Plugins are listed regardless of language.
+    repeated PluginLanguage languages = 3 [
+      (buf.validate.field).repeated.unique = true,
+      (buf.validate.field).repeated.items.enum.defined_only = true,
+      (buf.validate.field).repeated.items.enum.not_in = 0
+    ];
+    // The registry types to list Plugins for.
+    //
+    // If empty, all Plugins are listed regardless of registry type.
+    repeated PluginRegistryType registry_types = 4 [
+      (buf.validate.field).repeated.unique = true,
+      (buf.validate.field).repeated.items.enum.defined_only = true,
+      (buf.validate.field).repeated.items.enum.not_in = 0
+    ];
```

Comment:
Thoughts on adding a `name_query` (open to suggestions on the field name) to `Filter`, similar to [`ListLabels`](https://github.com/bufbuild/registry-proto/blob/c4103f09d463d36c2cc33816b3e1e2d951a915e6/buf/registry/module/v1/label_service.proto#L139-L140)? This would allow for a client to fuzzy-filter based on user input.

Reply:
Oh, is that what we're using it for? If I input `protoc-gen-` it'll match `protoc-gen-go-*`?

Reply:
yeah — I was thinking of a user entering `grp` and seeing all the plugins with the `grpc/` owner (hence maybe not using `name_query`, unless we consider "owner" as a part of the plugin's "name"), or e.g. `community/scalapb-zio-grpc`. Similar to the experience on https://buf.build/plugins now, but server-side instead of client-side.

Reply:
Gotcha, I feel like filtering plugins by organizations is covered by the `owner_refs` filter. Although you have to supply the exact organization name (or id).

I supposed we could have a `fullname_query` that is orthogonal to filtering by an organization, which could support:

- `grpc/*` (already covered by `owner_refs`)
- `grp*`
- `gatewa*`
- `json*`

I think these would be the most common types of searches. And searching by plugin name is typically what you'd want?

So maybe `name_query`, which applies to just plugin names, is the most appropriate for most cases?